rollup/circuitcapacitychecker/libzkp/src/lib.rs

```diff
@@ -66,13 +66,11 @@ pub mod checker {
             let tx_traces_vec = c_char_to_vec(tx_traces);
             let traces = serde_json::from_slice::<BlockTrace>(&tx_traces_vec)
                 .unwrap_or_else(|_| panic!("id: {id:?}, fail to deserialize tx_traces"));
-            if traces.transactions.len() != 1 {
-                panic!("traces.transactions.len() != 1")
-            } else if traces.execution_results.len() != 1 {
-                panic!("traces.execution_results.len() != 1")
-            } else if traces.tx_storage_trace.len() != 1 {
-                panic!("traces.tx_storage_trace.len() != 1")
-            }
+
+            assert_eq!(traces.transactions.len(), 1, "traces.transactions.len() != 1");
```

Comment:
Assert will still panic, no? But the main concern here is the panic after `estimate_circuit_capacity`

Reply:
I misunderstood, I thought this PR just for better codes.  I will look into it.

Reply:
We still want to keep `panic_unwind` but we don't want to panic on expected errors. So we should just return the error from `estimate_circuit_capacity` instead of panicking.

Reply:
> We still want to keep `panic_unwind` but we don't want to panic on expected errors. So we should just return the error from `estimate_circuit_capacity` instead of panicking.

For `return the error`, where should we return it? should we return the error to golang? now these are all rust codes.  My opinion is logging  these `estimate_circuit_capacity` errors. what do you think?

Reply:
It's already returned here, no need to log.

https://github.com/scroll-tech/go-ethereum/blob/ba76f768ce3a9907327d486b645f911be489dff4/rollup/circuitcapacitychecker/libzkp/src/lib.rs#L105